pandas/io/tests/test_pickle.py

```diff
@@ -284,6 +284,46 @@ def test_pickle_v0_15_2(self):
         #
         tm.assert_categorical_equal(cat, pd.read_pickle(pickle_path))
 
+    def compression_explicit(self, compression):
+        # issue 11666
+        if compression == 'xz':
+            tm._skip_if_no_lzma()
+        with tm.ensure_clean(self.path) as path:
+            df = tm.makeDataFrame()
+            df.to_pickle(path, compression=compression)
+            df2 = pd.read_pickle(path, compression=compression)
+            tm.assert_frame_equal(df, df2)
+
+    def test_compression_explicit(self):
+        compressions = [None, 'gzip', 'bz2', 'xz']
```

Comment:
Not this version, maybe next version. Actually in most cases I don't think it's a good design to compress one pickle file into `zip` file and then enforce there is only one file in that `zip`. Maybe it's useful to compress multiple pickle files into one `zip` archive, but that's another story.